frodo.js

```diff
@@ -1,49 +1,45 @@
-const lastElement = array[array.length - 1];
```

Comment:
SCRIPT-I'm impressed with your use of design patterns here, it really enhances the code organization.